lms/djangoapps/learner_recommendations/utils.py

```diff
@@ -133,37 +194,33 @@ def get_amplitude_course_recommendations(user_id, recommendation_id):
     return True, False, []
 
 
-def filter_recommended_courses(user, unfiltered_course_keys, recommendation_count=5):
+def filter_recommended_courses(
+    user, unfiltered_course_keys, recommendation_count=5, user_country_code=None, request_course=None
+):
     """
     Returns the filtered course recommendations. The unfiltered course keys
     pass through the following filters:
-        1. Remove courses that a user is already enrolled in
-        2. <To be added later>
+        1. Remove courses that a user is already enrolled in.
+        2. If user is seeing the recommendations on a course about pages, filter that course out of recommendations.
+        3. Remove the courses which is restricted in user region.
 
     Returns:
-        filtered_recommended_courses (list): A list of course objects. Each item has
-        the following details for a course:
-            - course key (course_key)
-            - title
-            - partner image url (logo_image_url)
-            - marketing url for the course (marketing_url)
+        filtered_recommended_courses (list): A list of filtered course objects.
     """
     filtered_recommended_courses = []
-    fields = ["title", "owners", "marketing_url"]
+    fields = ["key", "uuid", "title", "owners", "image", "url_slug", "course_runs", "location_restriction"]
 
     # Remove the course keys a user is already enrolled in
-    enrollable_course_keys = _remove_user_enrolled_course_keys(user, unfiltered_course_keys)[:recommendation_count]
+    enrollable_course_keys = _remove_user_enrolled_course_keys(user, unfiltered_course_keys)
+    # If user is seeing the recommendations on a course about pages, filter that course out of recommendations
+    recommended_course_keys = [course_key for course_key in enrollable_course_keys if course_key != request_course]
+
+    for course_id in recommended_course_keys:
+        if len(filtered_recommended_courses) >= recommendation_count:
+            break
```

Comment:
we should not be doing this now considering we are checking for "active course runs" after the filtration step. If we do this here, we might end up with less courses if there are no active course run in the first 4-5 recommendations.

Reply:
We should not remove this from here instead we can add `user_country_code`  in if condition to optimize the dashboard case
```suggestion
        if user_country_code and len(filtered_recommended_courses) >= recommendation_count:
            break
```